comp/trace/agent/def/component.go

```diff
@@ -6,7 +6,25 @@
 // Package traceagent provides the agent component type.
 package traceagent
 
+import (
+	"context"
+	"net/http"
+
+	pb "github.com/DataDog/datadog-agent/pkg/proto/pbgo/trace"
+
+	"github.com/DataDog/opentelemetry-mapping-go/pkg/otlp/attributes"
+	"github.com/DataDog/opentelemetry-mapping-go/pkg/otlp/attributes/source"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
 // team: agent-apm
 
 // Component is the agent component type.
-type Component interface{}
+type Component interface {
+	// SetOTelAttributeTranslator sets the OTel attributes translator of the underlying trace agent.
+	SetOTelAttributeTranslator(attrstrans *attributes.Translator)
```

Comment:
This feels similarly hacky to the statsd setter. Ideally this could be handled without mutating the agent like we do with statsd.

Reply:
@dineshg13 and I talked a bit about the attributes translator before. This exists for a legacy reason and is not well structured right now, e.g. the attributes translator is in `traceagent.config` struct while it is not really a config. We have a backlog ticket to refactor & improve the OTLP receiver in general, and this will be part of that effort.